### PR TITLE
genai: fix potential nil pointer dereference in SendMessage

### LINKED ISSUE #54: Occasional panic in (*ChatSession).SendMessage(...) method
Version: `github.com/google/generative-ai-go v0.7.0`

The library may occasionally panic when calling the (*ChatSession).SendMessage(...) method:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x8 pc=0x93a60d7]

goroutine 1 [running]:
github.com/google/generative-ai-go/genai.(*ChatSession).SendMessage(0xc0001e1880, {0x9cfe300, 0xa4613c0}, {0xc00088a070, 0x1, 0x1})
  /pkg/mod/github.com/google/generative-ai-go@v0.7.0/genai/chat.go:42 +0x197
```

After looking into the implementation, this can happen when `(m *GenerativeModel) generateContent(...)` returns the `iter.merged` value when the `err` variable equals `iterator.Done`, having this error returned on the first iteration.

A quick fix in `(m *GenerativeModel) generateContent(...)` can be:

```go
_, err := iter.Next()
if err == iterator.Done {
  if iter.merged == nil {
    return nil, err
  }
  return iter.merged, nil
}
```

If this suggestion is accepted, I can create a pull request with the fix.

### PR DESCRIPTION
**Pull request by reugn**
Resolves #54.

### CODE REVIEW

File: genai/client.go [hunk 1]
```diff
@@ -146,7 +146,7 @@ func (m *GenerativeModel) generateContent(ctx context.Context, req *pb.GenerateC
 	}
 	for {
 		_, err := iter.Next()
-		if err == iterator.Done {
+		if err == iterator.Done && iter.merged != nil {
```
Comment (eliben) [inline genai/client.go:149]: I wonder if this is the right place to fix, or would that be in `Next` where `nil` is returned - which can happen without error only at the last line?

Comment (eliben) [reply genai/client.go:149]: @jba for thoughts on this

**Review by eliben — CHANGES_REQUESTED** [review]

File: genai/client.go (hunk 1 above)
Comment (jba) [reply genai/client.go:149]: I think we want to catch this even deeper. See #68.

**Comment by jba** [comment]
@reugn, although the fix is submitted, I still couldn't reproduce. Let us know if you still see a panic.

**Pull request closed without merging**